src/priceClient/adapters/across.ts

```diff
@@ -0,0 +1,60 @@
+import assert from "assert";
+import axios, { AxiosError } from "axios";
+import { msToS, PriceFeedAdapter, TokenPrice } from "../priceClient";
+
+type AcrossPrice = { price: number };
+const defaultTimeout = 5000; // mS
+
+export class PriceFeed implements PriceFeedAdapter {
+  protected _timeout = defaultTimeout;
+
+  constructor(public readonly name: string, public readonly host?: string) {
+    // Allow host to be overridden for test or alternative deployments.
+    if (this.host === undefined) this.host = "across.to";
+  }
+
+  get timeout(): number {
+    return this._timeout;
+  }
+
+  set timeout(timeout: number) {
+    assert(timeout >= 0);
+    this._timeout = timeout;
+  }
+
+  async getPriceByAddress(address: string, currency = "usd"): Promise<TokenPrice> {
+    // Note: Assume price is 60 seconds old since the API does not provide a timestamp.
+    const now = msToS(Date.now()) - 60;
+    const acrossPrice: AcrossPrice = await this.query(address, currency, this._timeout);
+
+    return { address: address, price: acrossPrice.price, timestamp: now };
+  }
+
+  // todo: Support bundled prices in the API endpoint.
+  async getPricesByAddress(addresses: string[], currency = "usd"): Promise<TokenPrice[]> {
+    const prices: TokenPrice[] = [];
+    for (const address of addresses) {
+      const price: TokenPrice = await this.getPriceByAddress(address, currency);
+      prices.push(price);
+    }
+
+    return prices;
+  }
```

Comment:
I think this code is a bit more idiomatic, and it also has the advantage of making the calls in parallel.

```suggestion
 getPricesByAddress(addresses: string[], currency = "usd"): Promise<TokenPrice[]> {
    return Promise.all(addresses.map(address => this.getPriceByAddress(address, currency)));
  }
```

Reply:
I'd thought about this, but wasn't sure whether it was a good idea to hit the same host _n_ times concurrently. For our load levels it's probably OK, but I think it'd be better if we updated the API to accept multiple prices in a single request.

I was also curious as to whether this might trigger ~_n_ parallel DNS lookups, but I tested it quickly and it looks like the DNS resolver cache in my local machine seems to recognise that it already has queries in flight and only makes a single request...so I think it's probably fine.

In any case, I've incorporated this change because I guess it'll be fine.